crates/sui-indexer/src/restorer/formal_snapshot.rs

```diff
@@ -0,0 +1,246 @@
+                )
+                .unwrap(),
+            ),
+        ));
+
+        Abortable::new(
+            async move {
+                let sema_limit = Arc::new(Semaphore::new(
+                    self.restore_config.object_store_concurrent_limit,
+                ));
+                let mut restore_tasks = vec![];
+
+                for (bucket, (part_num, file_metadata)) in input_files.into_iter() {
+                    let sema_limit_clone = sema_limit.clone();
+                    let epoch_dir_clone = epoch_dir.clone();
+                    let remote_object_store_clone = remote_object_store.clone();
+                    let sha3_digests_clone = sha3_digests.clone();
+                    let store_clone = self.store.clone();
+                    let bar_clone = move_object_progress_bar.clone();
+                    let restore_config = self.restore_config.clone();
+                    let next_checkpoint_after_epoch = self.next_checkpoint_after_epoch;
+
+                    let restore_task = task::spawn(async move {
+                        let _permit = sema_limit_clone.acquire().await.unwrap();
+                        let object_file_path = file_metadata.file_path(&epoch_dir_clone);
+                        let (bytes, _) = download_bytes(
+                            remote_object_store_clone,
+                            &file_metadata,
+                            epoch_dir_clone,
+                            sha3_digests_clone,
+                            &&bucket,
+                            &part_num,
+                            Some(restore_config.object_store_max_timeout_secs),
+                        )
+                        .await;
+                        info!(
+                            "Finished downloading move object file {:?}",
+                            object_file_path
+                        );
+                        let mut move_objects = vec![];
+                        let _result: Result<(), anyhow::Error> =
+                            LiveObjectIter::new(&file_metadata, bytes.clone()).map(|obj_iter| {
+                                for object in obj_iter {
+                                    match object {
+                                        LiveObject::Normal(obj) => {
+                                            if !obj.is_package() {
+                                                let indexed_object = IndexedObject::from_object(
+                                                    next_checkpoint_after_epoch,
+                                                    obj,
+                                                    None,
+                                                );
+                                                move_objects.push(indexed_object);
+                                            }
+                                        }
+                                        LiveObject::Wrapped(_) => {}
```

Comment:
Just want to make sure I understand -- we are not worrying about this because there will be no indexer snapshots from the period when we included wrapped tombstones in the live object set?

Reply:
@williampsmith yeah, my understanding is that the formal snapshot doesn't contain information about tombstones, only the live object set, as such wrapped objects shouldn't ever appear in formal snapshots. Is this accurate or is our understanding of the formal snapshot system flawed?

Reply:
Well to be clear, the formal snapshot contains whatever `iter_live_object_set` deems as a live object. There is an older version of that iter, and a corresponding version of state accumulator, which did include wrapped objects. But we have not been doing that for many months. So yeah this should be fine assuming we are not restoring from very old formal snapshots (incidentally, our Mysten snapshot hosting don't have anywhere near the retention to go back that far anyway)

Reply:
Perfect thanks for the clarity

Reply:
If we wanted to be more bulletproof in this regard though (for example in case someone were to want to replay from genesis and create a snapshot), we may need to cover that case. Easiest might be to prevent any such footguns by having some check here on the protocol config.

However, there's a circular dependency during snapshot restore, since we need to have an epoch store to get the protocol version. But since the epoch that this was introduced on each chain is static, you can do a hack like we do in https://github.com/MystenLabs/sui/blob/ab76bb5fe97895b5550cf0c7bcc2d9bd579ebe0c/crates/sui-snapshot/src/lib.rs#L263-L271. Basically if `include_tombstones` as a result of this logic block is `true`, you may wanna say indexer formal snapshots are unsupported, or just include them in the same way that `iter_live_object_set` does. cc @gegaowp

Reply:
Well we don't need the tombstones even if they were there, so I think just ignoring this now should be fine?